langchain/vectorstores/weaviate.py

```diff
@@ -150,15 +150,34 @@ def max_marginal_relevance_search(
         Returns:
             List of Documents selected by maximal marginal relevance.
         """
-        lambda_mult = kwargs.get("lambda_mult", 0.5)
-
         if self._embedding is not None:
             embedding = self._embedding.embed_query(query)
         else:
             raise ValueError(
                 "max_marginal_relevance_search requires a suitable Embeddings object"
             )
 
+        return self.max_marginal_relevance_search_by_vector(
+            embedding, k, fetch_k, **kwargs
+        )
+
+    def max_marginal_relevance_search_by_vector(
+        self, embedding: List[float], k: int = 4, fetch_k: int = 20, **kwargs: Any
+    ) -> List[Document]:
+        """Return docs selected using the maximal marginal relevance.
+
+        Maximal marginal relevance optimizes for similarity to query AND diversity
+        among selected documents.
+
+        Args:
+            embedding: Embedding to look up documents similar to.
+            k: Number of Documents to return. Defaults to 4.
+            fetch_k: Number of Documents to fetch to pass to MMR algorithm.
+
+        Returns:
+            List of Documents selected by maximal marginal relevance.
+        """
+        lambda_mult = kwargs.get("lambda_mult", 0.5)
```

Comment:
if this is something we'd reasonably expect users to change then maybe worth putting in function signature explicitly and adding to docstring

Reply:
I would be in favour of making the parameter of the algorithm explicit. Ideally, the method signature in the abstract `VectorStore` class would be updated and the change propagated to its implementations. It would have to then be mentioned in the documentation and examples. Any objections?

Reply:
we can just add the kwarg here, no? the way `Chroma.max_marginal_relevance_search_by_vector` has a `filter` kwarg for example

Reply:
I'll do that but I think in the future we should have consistency across all vector stores implementations.

Reply:
done

Reply:
I'm not sure I understand the problem. What is inconsistent with the approach you've chosen?

Reply:
I see. I understand now. I believe we should try to make sure that this stuff makes sense. People are going to run into that running the framework, and will have to make an issue an report it, then someone else will have to make a PR, etc. 

`lambda_mult = kwargs.get("lambda_mult", 0.5)` <- if we delete this in one place, we should delete it in all places. And then deal with the consequences of that action. If that means making this bubble up all the way to the abstract, so be it. I don't see a better way either. Just for dev experience alone, I would want important tweaking vars like this to be as explicit as possible so really it's just making it better in the first place anyway.

Reply:
I am happy to tidy that up if there are no objections from the community.

Reply:
I'll take the silence to mean no objection, so I'll go ahead and make the param explicit in the base class and update relevant code accordingly.

Reply:
done